packages/react-devtools-shared/src/backend/fiber/renderer.js

```diff
@@ -2540,7 +2594,7 @@ export function attach(
         updateFiberRecursively(
           nextFallbackChildSet,
           prevFallbackChildSet,
-          nextFiber,
+          newParentInstance, // Note this used to be the equivalent of fiberInstance if it was filtered but I think that was a bug.
```

Comment:
I believe this is a bug fix because it shouldn't pass nextFiber as the parent if this node was filtered out.

The old code should've been `shouldIncludeInTree ? nextFiber : parentFiber`.

Reply:
Looks like a bug, I've shipped a fix some time ago for mounting case - https://github.com/facebook/react/pull/27147, which had `shouldIncludeInTree ? nextFiber : parentFiber` logic as well